### PR TITLE
QR code background gradient

### PR DESCRIPTION
**Pull request by Pau1fitz**
Related to #18452

Add gradient background to QR code based on updated design. 

https://github.com/status-im/status-mobile/assets/6873736/60aee83c-ac3a-485f-a61c-0b65778eaebe

iOS
Android

Testing notes:
Navigate to the share screen
See updated qr-code with background gradient
click on Wallet
See updated qr-code with background gradient

### CODE REVIEW

File: src/quo/components/share/share_qr_code/view.cljs [hunk 1]
```diff
@@ -211,3 +216,4 @@
                (-> props
                    (assoc :component-width @component-width)
                    (clojure.set/rename-keys {:type :share-qr-type}))]))]]))
+
```
Comment (OmarBasem) [inline src/quo/components/share/share_qr_code/view.cljs:209]: Empty new line

File: src/quo/components/share/share_qr_code/view.cljs [hunk 2]
```diff
@@ -126,39 +128,42 @@
   [{:keys [share-qr-type qr-image-uri component-width customization-color full-name
            profile-picture emoji on-share-press]
     :as   props}]
-  [rn/view {:style style/content-container}
-   [rn/view
-    {:style style/share-qr-container}
+  [linear-gradient/linear-gradient
+   {:colors [(colors/resolve-color customization-color quo.theme/with-theme 15) :transparent]
```
Comment (OmarBasem) [inline src/quo/components/share/share_qr_code/view.cljs]: The view component needs to be wrapped with quo.theme and and pass the `theme` prop to `colors/resolve-color`. You can see example here: https://github.com/status-im/status-mobile/blob/develop/src/quo/components/wallet/network_routing/view.cljs#L189C1-L189C1

File: src/quo/components/share/share_qr_code/view.cljs (hunk 1 above)
Comment (Pau1fitz) [reply src/quo/components/share/share_qr_code/view.cljs:209]: resolved 👍

File: src/quo/components/share/share_qr_code/view.cljs (hunk 2 above)
Comment (Pau1fitz) [reply src/quo/components/share/share_qr_code/view.cljs]: I have updated to the code to wrap the outer view with `quo.theme` and updated the name to `view-internal`. Let me know if this is the preferred approach, or if we want to wrap the inner component?

Comment (J-Son89) [reply src/quo/components/share/share_qr_code/view.cljs]: Looks good as is 👌

File: src/quo/components/share/share_qr_code/view.cljs [hunk 3]
```diff
@@ -124,43 +126,46 @@
...
+       :emoji               emoji}]]
+    [rn/view {:style style/bottom-container}
+     (case share-qr-type
+       :profile           [profile-bottom props]
+       :wallet-legacy     [wallet-legacy-bottom props]
+       :wallet-multichain [wallet-multichain-bottom props]
+       nil)]]])
 
-(defn view
+(defn view-internal
```
Comment (OmarBasem) [inline src/quo/components/share/share_qr_code/view.cljs]: Please make the internal view private `defn-`

Comment (Pau1fitz) [reply src/quo/components/share/share_qr_code/view.cljs]: done @OmarBasem 👍

**Review by OmarBasem — APPROVED** [review]

**Review by J-Son89 — APPROVED** [review]

**Comment by J-Son89** [comment]
@Francesca-G can you check this looks alright when you get a chance?

**Review by Francesca-G — COMMENTED** [review]
Here's the Figma [frame](https://www.figma.com/file/Tf5nfkYvpbnNCo4rKLK7lS/Feedback-for-Mobile?type=design&node-id=8521%3A7056&mode=design&t=mk3q6T03QPBcW7c9-1) with the design review :)

I'm also attaching a video about the color contamination and gradient uniformity issues I've encountered while scrolling:

https://github.com/status-im/status-mobile/assets/122543089/799ca20d-1a09-422e-8d1d-15f24d8c40a1

**Comment by Pau1fitz** [comment]
@Francesca-G I am unable to view the figma frame. Would it be possible to be granted access?

**Review by Francesca-G — APPROVED** [review]

**Comment by J-Son89** [comment]
@status-im/mobile-qa this pr is okay to merge? it's only changing ui on the share screen and e2e tests ran with 92% https://github.com/status-im/status-mobile/pull/18503#issuecomment-1897117125

**Comment by pavloburykh** [comment]
> @status-im/mobile-qa this pr is okay to merge? it's only changing ui on the share screen and e2e tests ran with 92% [#18503 (comment)](https://github.com/status-im/status-mobile/pull/18503#issuecomment-1897117125)

@J-Son89 yes, we can merge it. Just to clarify: this PR only adds gradient and [indicators](https://github.com/status-im/status-mobile/issues/18452#issuecomment-1893432659) will be added in separate PR, correct?

**Comment by Pau1fitz** [comment]
@pavloburykh that is correct. The indicators will be added in[ this PR](https://github.com/status-im/status-mobile/pull/18523) which is currently in review.

**Pull request merged**